File: 프로그래머스/unrated/138476. 귤 고르기/귤 고르기.py
from collections import defaultdict

def solution(k, tangerine):
    dd = defaultdict(int)
    for i in tangerine:
        dd[i] += 1

    dd = sorted(dd.values(), reverse = True)

    answer = 0
    for i in range(len(dd)):
        answer += 1
        k -= dd[i]
        if k <= 0:
            break
    return answer

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.swift
import Foundation

func solution(_ k:Int, _ tangerine:[Int]) -> Int {
    var K = k
    var dic = [Int: Int]()
    for i in tangerine {
        dic[i, default:0] += 1
    }
    var sortedDic = dic.values.sorted(by: {$0 > $1})
    
    var answer = 0
    for i in 0..<sortedDic.count {
        answer += 1
        K -= sortedDic[i]
        if K <= 0 {
            break
        }
    }
    return answer
}